include/CL/sycl/h_item.hpp

```diff
@@ -34,12 +34,12 @@ namespace cl {
 namespace sycl {
 
 template<int dimensions>
-class group;
+struct group;
```

Comment:
I've never seen any warnings related to this forward declaration. What's the problem with forward declaring classes?

Reply:
It's not so much about being a `class`, but the actual declaration of `group` being a `struct`. The warning is `warning: 'group' defined as a struct template here but previously declared as a class template; this is valid, but may result in linker errors under the Microsoft C++ ABI`

Reply:
(here references the actual definition and the previous declaration is the forward declaration)

Reply:
Okay, I see. That makes sense.